File: src/main/java/ru/olejkai/task_vsr/payload/request/SignupRequest.java
package ru.olejkai.task_vsr.payload.request;

import lombok.Data;
import ru.olejkai.task_vsr.annotation.PasswordMatchers;

import javax.validation.constraints.NotEmpty;

@PasswordMatchers(message = "wrong confirmPassword")
@Data
public class SignupRequest {





    @NotEmpty(message = "tabelNumber not empty")
    private String tabelNumber;
    /*@NotEmpty(message = "username not empty")
    private  String username;*/
    @NotEmpty(message = "name not empty")
    private  String name;
    @NotEmpty(message = "surname not empty")
    private String  surname;
    private String telephoneNumber;
    @NotEmpty(message = "postHasDepartmentId not empty")
    private String postHasDepartmentId;
    @NotEmpty(message = "password not empty")
    private  String password;
    private  String confirmPassword;

}

File: frontend/src/app/models/SignupRequest.ts
export class SignupRequest{
  tabelNumber:string;
  name:string;
  surname:string;
  telephoneNumber:string;
  postHasDepartmentId:string;
  password:string;
  confirmPassword:string;


  constructor(tabelNumber: string, name: string, surname: string, telephoneNumber: string, postHasDepartmentId: string, password: string, confirmPassword: string) {
    this.tabelNumber = tabelNumber;
    this.name = name;
    this.surname = surname;
    this.telephoneNumber = telephoneNumber;
    this.postHasDepartmentId = postHasDepartmentId;
    this.password = password;
    this.confirmPassword = confirmPassword;
  }
}

